### PR TITLE
Make gcp benchmark dependencies mockable

### PR DESCRIPTION
**Pull request by uri-weisman**
### Summary of your changes
1. Shift gcp auth util and the inventory service to become a dependency in order to be mockable.
2. Modify the credential util to support the case in which no credentials have been provided.
As a result, the GCP SDK will use ADC (application default creds) and will use the compute instance's SA.

### Related Issues
- Blocked by: https://github.com/elastic/cloudbeat/pull/1169 (built on top of this branch)

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

**Review by orestisfl — APPROVED** [review]

**Comment by oren-zohar** [comment]
@uri-weisman test coverage decreased?

File: flavors/benchmark/benchmark.go [hunk 1]
```diff
@@ -69,7 +71,9 @@ func NewBenchmark(cfg *config.Config) (Benchmark, error) {
 		}, nil
 	case config.CIS_GCP:
 		return &GCP{
-			IdentityProvider: &identity.Provider{},
+			IdentityProvider:     &identity.Provider{},
+			CfgProvider:          &auth.ConfigProvider{},
+			inventoryInitializer: &inventory.ProviderInitializer{},
```
Comment (orestisfl) [inline flavors/benchmark/benchmark.go:76]: (not actionable in this PR observation) We don't really need `GCP` or any other `Benchmark` implementation to be exported. Until we actually need to do that, we can make everything private.

**Comment by uri-weisman** [comment]
> @uri-weisman test coverage decreased?

@oren-zohar less than 0.1% percent 🙈

**Comment by oren-zohar** [comment]
@uri-weisman I just mentioned this since it is counterintuitive, as this PR is around making more testing possible, no?

**Comment by uri-weisman** [comment]
> @uri-weisman I just mentioned this since it is counterintuitive, as this PR is around making more testing possible, no?

This PR was all about making the utility's API calls mockable, so they wouldn't actually run. Though we wanted to improve code coverage, this PR didn't have many logic changes. The reduction in coverage was mainly due to shifts in code blocks.

**Pull request merged**